sales_billing/sales_billing/doctype/sales_billing/sales_billing.py

```diff
@@ -52,3 +52,97 @@ def update_sales_billing_outstanding_amount(doc, method):
         total_outstanding_amount += invoice.outstanding_amount
     bill.total_outstanding_amount = total_outstanding_amount
     bill.save()
+
+@frappe.whitelist()
+def create_payment_receipt(payment_details, sales_billing_name, posting_date):
+    import json
+    try:
+        payment_details = json.loads(payment_details)
+    except Exception:
+        frappe.throw(_("Failed to parse payment details. Please check your input."))
+
+    sales_billing = frappe.get_doc('Sales Billing', sales_billing_name)
+    customer, company = sales_billing.customer, sales_billing.company
+    company_currency = frappe.get_value('Company', company, 'default_currency')
+    total_outstanding_amount = sales_billing.total_outstanding_amount
+    total_paid_amount = sum(detail['paid_amount'] for detail in payment_details)
```

Comment:
```suggestion
    total_paid_amount = sum([detail['paid_amount'] for detail in payment_details])
```
I can't see why there is not eeror if len(payment_details) > 0. I think input to sum() should be a list